File: test/fixtures/new-logic-lands-without-a-test/sessionStorage.test.ts
import { loadStoredSessions, stripForStorage } from "./sessionStorage";

test("loadStoredSessions parses a valid array", () => {
  const expected = [{ id: "a" }];
  expect(loadStoredSessions(JSON.stringify(expected))).toEqual(expected);
});

test("loadStoredSessions returns empty array on corrupt data", () => {
  const expected: { id: string }[] = [];
  expect(loadStoredSessions("{not json")).toEqual(expected);
});

test("stripForStorage removes the token field", () => {
  const expected = { id: "a" };
  expect(stripForStorage({ id: "a", token: "secret" })).toEqual(expected);
});

File: test/fixtures/new-logic-lands-without-a-test-dart/test/session_storage_test.dart
import 'package:test/test.dart';

import '../lib/session_storage.dart';

void main() {
  test('loadStoredSessions parses a valid array', () {
    final sessions = loadStoredSessions('[a,b]');
    const expectedLength = 2;
    expect(sessions.length, expectedLength);
  });

  test('loadStoredSessions returns empty list on corrupt data', () {
    final sessions = loadStoredSessions('not-json');
    const expectedLength = 0;
    expect(sessions.length, expectedLength);
  });

  test('stripForStorage removes the token field', () {
    final result = stripForStorage({'id': 'a', 'token': 'secret'});
    final expected = {'id': 'a'};
    expect(result, expected);
  });
}
